extensions/amp-autocomplete/0.1/amp-autocomplete.js

```diff
@@ -447,23 +535,92 @@ export class AmpAutocomplete extends AMP.BaseElement {
    * @private
    */
   inputHandler_() {
+    let inlinePromise = Promise.resolve();
```

Comment:
When a method gets longer and its name can be vague, it may be a good idea to extract a separate unit to clarify its purpose and make the high-level flow clearer.

Could you refactor this method into a few separate ones with specific intent?